scripts/stubsabot.py

```diff
@@ -498,32 +500,50 @@ async def create_or_update_pull_request(*, title: str, body: str, branch_name: s
+        f"{TYPESHED_API_URL}/pulls",
         json={"title": title, "body": body, "head": f"{fork_owner}:{branch_name}", "base": "main"},
         headers=get_github_api_headers(),
     ) as response:
         resp_json = await response.json()
-        if response.status == 422 and any(
+        if response.status == 201:
+            pr_number = resp_json["number"]
+            assert isinstance(pr_number, int)
+        elif response.status == 422 and any(
             "A pull request already exists" in e.get("message", "") for e in resp_json.get("errors", [])
         ):
-            # Find the existing PR
-            async with session.get(
-                f"https://api.github.com/repos/{TYPESHED_OWNER}/typeshed/pulls",
-                params={"state": "open", "head": f"{fork_owner}:{branch_name}", "base": "main"},
-                headers=get_github_api_headers(),
-            ) as response:
-                response.raise_for_status()
-                resp_json = await response.json()
-                assert len(resp_json) >= 1
-                pr_number = resp_json[0]["number"]
-            # Update the PR's title and body
-            async with session.patch(
-                f"https://api.github.com/repos/{TYPESHED_OWNER}/typeshed/pulls/{pr_number}",
-                json={"title": title, "body": body},
-                headers=get_github_api_headers(),
-            ) as response:
-                response.raise_for_status()
-            return
+            pr_number = await update_existing_pull_request(title=title, body=body, branch_name=branch_name, session=session)
+        else:
+            response.raise_for_status()
+            raise AssertionError(f"Unexpected response: {response.status}")
+    await update_pull_request_label(pr_number=pr_number, session=session)
+
+
+async def update_existing_pull_request(*, title: str, body: str, branch_name: str, session: aiohttp.ClientSession) -> int:
+    fork_owner = get_origin_owner()
+
+    # Find the existing PR
+    async with session.get(
+        f"{TYPESHED_API_URL}/pulls",
+        params={"state": "open", "head": f"{fork_owner}:{branch_name}", "base": "main"},
+        headers=get_github_api_headers(),
+    ) as response:
+        response.raise_for_status()
+        resp_json = await response.json()
+        assert len(resp_json) >= 1
+        pr_number = resp_json[0]["number"]
+        assert isinstance(pr_number, int)
+    # Update the PR's title and body
+    async with session.patch(
+        f"{TYPESHED_API_URL}/pulls/{pr_number}", json={"title": title, "body": body}, headers=get_github_api_headers()
+    ) as response:
+        response.raise_for_status()
+    return pr_number
+
+
+async def update_pull_request_label(*, pr_number: int, session: aiohttp.ClientSession) -> None:
+    async with session.post(
+        f"{TYPESHED_API_URL}/issues/{pr_number}/labels", json={"labels": [STUBSABOT_LABEL]}, headers=get_github_api_headers()
```

Comment:
```suggestion
        f"{TYPESHED_API_URL}/pulls/{pr_number}/labels", json={"labels": [STUBSABOT_LABEL]}, headers=get_github_api_headers()
```

Reply:
That didn't work when I tried it. At least with GET.

Reply:
Oh, fair enough. https://github.com/python/typeshed/pull/10506 looked pretty convincing, so I guess this works as-is. Maybe worth adding a comment, though? It _looks_ incorrect, even if it isn't 😄